src/partitioning/bvh.rs

```diff
@@ -135,12 +138,15 @@ pub trait BVH<T, BV> {
                     match visitor.visit(best_cost, child_bv, child_data) {
                         BestFirstVisitStatus::Continue { cost, result } => {
                             if cost < best_cost {
+                                // result is_some iff child is a leaf node
                                 if result.is_some() {
+                                    // This is the nearest collision so far
                                     best_cost = cost;
                                     best_result = result.map(|res| (child, res));
+                                } else {
+                                    // BV may have a child with lower cost, evaluate it next.
+                                    queue.push(WeightedValue::new(child, -cost))
```

Comment:
This should not be an `else` case. The `queue.push` should always happen just like before this change because we could allow non-leaf nodes to contain non-None results in the future.